Framework/Core/src/O2ControlHelpers.cxx

```diff
@@ -9,106 +9,327 @@
-        continue;
+  }
+  return rawChannels;
+}
+
+void dumpCommand(std::ostream& dumpOut, const DeviceExecution& execution, std::string indLevel)
+{
+  dumpOut << indLevel << "shell: true\n";
+  dumpOut << indLevel << "log: \"{{ log_task_output }}\"\n";
+  dumpOut << indLevel << "user: \"{{ user }}\"\n";
+
+  dumpOut << indLevel << "value: >-\n";
+  // fixme : how to find out when to include QC? Now we include it always, just in case
+  dumpOut << indLevel << indScheme << "source /etc/profile.d/modules.sh && MODULEPATH={{ modulepath }} module load O2 QualityControl Control-OCCPlugin &&\n";
+  // fixme: For now, we will use a dump file, but to further automatise it
+  //  I need the full command that was used before merging the workflows,
+  //  so it is run each time. (on the other hand, it may slow down the start up).
+  // fixme: trim the path to the exec itself when needed (not /home/username/bla/o2-datasampling)
+  dumpOut << indLevel << indScheme << "cat {{ dpl_config }} | " << execution.args[0] << "\n";
+
+  dumpOut << indLevel << "arguments:\n";
+  dumpOut << indLevel << indScheme << "- \"-b\"\n";
+  dumpOut << indLevel << indScheme << "- \"--monitoring-backend\"\n";
+  dumpOut << indLevel << indScheme << "- \"'{{ monitoring_dpl_url }}'\"\n";
+  dumpOut << indLevel << indScheme << "- \"--session\"\n";
+  dumpOut << indLevel << indScheme << "- \"'default'\"\n";
+  dumpOut << indLevel << indScheme << "- \"--infologger-severity\"\n";
+  dumpOut << indLevel << indScheme << "- \"'{{ infologger_severity }}'\"\n";
+  dumpOut << indLevel << indScheme << "- \"--infologger-mode\"\n";
+  dumpOut << indLevel << indScheme << "- \"'{{ infologger_mode }}'\"\n";
+  dumpOut << indLevel << indScheme << "- \"--driver-client-backend\"\n";
+  dumpOut << indLevel << indScheme << "- \"'stdout://'\"\n";
+  dumpOut << indLevel << indScheme << "- \"--shm-segment-size\"\n";
+  dumpOut << indLevel << indScheme << "- \"'{{ shm_segment_size }}'\"\n";
+  dumpOut << indLevel << indScheme << "- \"--shm-throw-bad-alloc\"\n";
+  dumpOut << indLevel << indScheme << "- \"'{{ shm_throw_bad_alloc }}'\"\n";
+
+  for (size_t ai = 1; ai < execution.args.size(); ++ai) {
+    const char* option = execution.args[ai];
+    const char* value = nullptr; // no value by default (i.e. a boolean)
+    // If the subsequent option exists and does not start with -, we assume
+    // it is an argument to the previous one.
+    // ...that is unless it is a "-1" for example.
+    if (ai + 1 < execution.args.size() && execution.args[ai + 1][0] != '-') {
+      value = execution.args[ai + 1];
+      ai++;
+    }
+    if (!option) {
+      break;
+    }
+
+    static const std::set<std::string> omitOptions = {
+      "--channel-config", "--o2-control", "--control", "--session", "--monitoring-backend",
+      "-b", "--color", "--infologger-severity", "--infologger-mode", "--driver-client-backend",
+      "--shm-segment-size", "--shm-throw-bad-alloc"};
```

Comment:
@ktf @teo  Can you think of any other options that should be omitted or replaced by AliECS?
Here you can find examples of the full argument lists https://github.com/knopers8/ControlWorkflows/commit/bc5e3590c7d25d9f1eedadbeea8e4641fe1e7ef9